### PR TITLE
A-Z/a-z for DimElement

### LINKED ISSUE #212: dimIndexType support for [A-Z]-[A-Z] pattern
The CMSIS SVD [v1.3.9 specification](https://www.keil.com/pack/doc/CMSIS/SVD/html/schema_1_2_gr.html) describes the valid patterns for `dimIndexType` as `"[0-9]+\-[0-9]+|[A-Z]-[A-Z]|[_0-9a-zA-Z]+(,\s*[_0-9a-zA-Z]+)+"`. When parsing R7FA6M5BH.svd from vendor Renesas found [here](https://www2.renesas.eu/Keil_MDK_Packs/Renesas.RA_DFP.4.0.0.pack), svd2rust throws the following error: "invalid digit found in string".

Requesting support for the full `dimIndexType` specification, thanks!

### PR DESCRIPTION
**Pull request by burrbull**
Should fix #212

cc @n8tlarsen Test, please

### CODE REVIEW

**Comment by n8tlarsen** [comment]
Adding the following patch to the svd2rust/master Cargo.toml causes numerous build errors. Could you try the same and take a look?
``` toml
[patch.crates-io]
svd-parser = { git = "https://github.com/rust-embedded/svd.git", branch = "indices" }
```

**Comment by burrbull** [comment]
You need also change 
`[dependencies.svd-rs]`

**Comment by n8tlarsen** [comment]
Got all the tools built. It works for one example SVD I found which specified "A-B" for the DimElement.

**Comment by burrbull** [comment]
cc @Emilgardis

File: svd-rs/src/dimelement.rs [hunk 1]
```diff
@@ -144,15 +144,26 @@ impl DimElement {
...
+            if let (Ok(start), Ok(end)) = (start.parse::<u32>(), end.parse::<u32>()) {
+                Some((start..=end).map(|i| i.to_string()).collect::<Vec<_>>())
+            } else {
+                let mut start = start.bytes();
+                let mut end = end.bytes();
+                match (start.next(), start.next(), end.next(), end.next()) {
+                    (Some(start), None, Some(end), None)
+                        if start.is_ascii_alphabetic() && end.is_ascii_alphabetic() =>
+                    {
+                        Some((start..=end).map(|c| char::from(c).to_string()).collect())
```
Comment (Emilgardis) [inline svd-rs/src/dimelement.rs:158]: this allows `A-z` which is not in the spec, is that ok?

File: svd-rs/src/dimelement.rs [hunk 2]
```diff
@@ -144,15 +144,26 @@ impl DimElement {
...
     /// Get array of indexes from string
     pub fn parse_indexes(text: &str) -> Option<Vec<String>> {
-        if text.contains('-') {
-            let mut parts = text.splitn(2, '-');
-            let start = parts.next()?.parse::<u32>().ok()?;
-            let end = parts.next()?.parse::<u32>().ok()?;
-
-            Some((start..=end).map(|i| i.to_string()).collect())
+        (if text.contains('-') {
+            let (start, end) = text.split_once('-')?;
```
[the remark this answers is not in the document]
Comment (Emilgardis) [reply svd-rs/src/dimelement.rs:148]: altough, the condensed version has some faults when the data is malformed, so maybe not good to do

File: svd-rs/src/dimelement.rs (hunk 1 above)
Comment (burrbull) [reply svd-rs/src/dimelement.rs:159]: Fixed

**Review by Emilgardis — APPROVED** [review]
bors r+

nice!

**Pull request merged**